src/rat/core/sarea/bot_filter.py

```diff
@@ -0,0 +1,236 @@
+    which filters the optical derived surface area with SAR surface area as the reference.
+    
+    Parameters
+    ----------
+    tmsos_gee_path : str
+        The path to the tmsos surface area folder within the basin data directory
+    basin_reservoir_shpfile_path : str
+        The path to the file containing reservoir shapefile data
+    apply: boolean
+        Toggles whether the BOT filter is run. It is also used to reset the BOT Filtered surface area values to TMSOS if required (set to False).
+        Values: True/False
+    bias_threshold: float
+        Controls the filtering intensity for the bias between Optical and SAR surface areas.
+        Values: 0 - 9
+    outlier_threshold: float
+        Controls the intensity of filtering out the outliers from the Optical surface area data.
+        Values: 0 - 9
+    trend_threshold: float
+        Controls the intensity of filtering the Optical surface area data based on the deviation in trends from the SAR surface area data.
+        Values: 0 - 9
+    '''
+   
+    #  Reading the reservoir shapefile data and creating the list of reservoirs in the basin. 
+    basin_reservoir_data = gpd.read_file(basin_reservoir_shpfile_path)
+    log_level1.debug(basin_reservoir_shpfile_path)
+    res_list = [row["DAM_NAME"] if pd.isna(row["GRAND_ID"]) else str(int(row["GRAND_ID"])) +'_'+ row["DAM_NAME"] for index, row in basin_reservoir_data.iterrows()]
+        
+    if(apply == True):
+        log_level1.info('Running BOT filter for Surface Area')
+        res_nomArea = basin_reservoir_data["AREA_SKM"]
+        # Scaling filtering thresholds to actual filtering values. If threshold value is 0, they are assinged an extremely high value to essentially turn it off.
+        # sigma from monthly mean; Mapped from (1-9 to 4-0.5sigma)
+        filt_1_thresh = (9999 if(outlier_threshold==0) else ((3.5/8 * (9 - outlier_threshold)) + 0.5)) 
+        # % of nom surface area deviation from SAR; Mapped from (1-9 to 20-5% )
+        filt_2_thresh = (9999 if(bias_threshold==0) else ((15/8 * (9 - bias_threshold)) + 5)) 
+        # monthly sigma deviation from SAR trend(avg of prev 2 sar trends); Mapped from (1-9 to 3-0.1sigma)
+        filt_3_thresh = (9999 if(trend_threshold==0) else ((2.9/8 * (9 - trend_threshold)) + 0.1)) 
+        # Initialising counter for failures
+        count_failed = 0
+        botFilter_status = pd.DataFrame(columns=['Reservoir', 'Status'])
+        for curr_dam_index,res_name in enumerate(res_list):   
+            try:
+                res_wa_timeseries_l8 =  pd.read_csv(os.path.join(tmsos_gee_path,'l8',res_name +'.csv'))
+                res_wa_timeseries_s2 =  pd.read_csv(os.path.join(tmsos_gee_path,'s2',res_name +'.csv'))
+                res_wa_timeseries_s1 =  pd.read_csv(os.path.join(tmsos_gee_path,'sar',res_name +'_12d_sar.csv'))
+                res_wa_timeseries_l9 =  pd.read_csv(os.path.join(tmsos_gee_path,'l9',res_name +'.csv'))
```

Comment:
There should be check if the file exists or not. Even if one satellite optical data is present, it should work and if SAR is there. Otherwise, raise exception.

Reply:
Also, create a check if SAR file is there otherwise raise exception.

Reply:
File checks added in new commit